gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/support/PubSubConfigurationFactory.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.pubsub.support;
+
+import com.google.api.gax.batching.BatchingSettings;
+import com.google.api.gax.batching.FlowControlSettings;
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.core.FixedExecutorProvider;
+import com.google.api.gax.grpc.InstantiatingGrpcChannelProvider;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.cloud.pubsub.v1.Publisher;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.gcp.GoogleCloudConfiguration;
+import io.micronaut.gcp.UserAgentHeaderProvider;
+import io.micronaut.gcp.pubsub.configuration.PubSubConfigurationProperties;
+import org.threeten.bp.Duration;
+
+import javax.inject.Singleton;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * Factory class to create default settings for PubSub Publisher and subscriber beans.
+ *
+ * @author Vinicius Carvalho
+ * @since 2.0.0
+ *
+ */
+@Factory
+@Requires(classes = Publisher.class)
+public class PubSubConfigurationFactory {
+
+    private final PubSubConfigurationProperties pubSubConfigurationProperties;
+
+    private final GoogleCloudConfiguration googleCloudConfiguration;
+
+    public PubSubConfigurationFactory(PubSubConfigurationProperties pubSubConfigurationProperties, GoogleCloudConfiguration googleCloudConfiguration) {
+        this.pubSubConfigurationProperties = pubSubConfigurationProperties;
+        this.googleCloudConfiguration = googleCloudConfiguration;
+    }
+
+    /**
+     * @return Default retrySettings
+     */
+    @Singleton
+    protected RetrySettings retrySettings() {
```

Comment:
We try to avoid this type of logic. You can likely setup Micronaut to bind to `RetrySettings.Builder` directly instead

Reply:
@jameskleeh I tried to fix this, but somehow I'm still getting a null value for the Builder settings. If you don't mind taking a look and see what's wrong. Thanks

Reply:
@viniciusccarvalho I've submitted a PR to your repo to fix the configuration

Reply:
@jameskleeh Thank you very much! Merged it, and will soon push latest changes. Thanks for the support